### PR TITLE
Sprint 6 comments

### PR DESCRIPTION
**Pull request by purplesmurf1998**
# Description
This PR lets users create/add comments to a vehicle and delete their comments from vehicle.
# How to test

1. Select a vehicle from an inventory
2. Navigate to the bottom right and click on "Write a comment"
3. Write your comment and format it the way you want to
4. Click "Post" to publish your comment
5. To delete the comment, simple press on the trash icon and confirm your request.
# Screenshots
image
image
image
image
# Jira Story
[[AP-20] As a user, I would like to comment on vehicles in the dealership](https://soen390-team12.atlassian.net/browse/AP-20?atlOrigin=eyJpIjoiMmRhNTI5NmM5ZDBkNDFjN2IxYzdiNzg2Y2UwOThmYmEiLCJwIjoiaiJ9)

### CODE REVIEW

File: frontend/src/views/pages/inventory/Inventory.vue [hunk 1]
```diff
@@ -39,121 +28,65 @@
...
+    showQueryParams() {
+      console.log(this.$route.query);
+    },
     showMessage(message) {
-      this.Message = message;
+      this.Message = message.message;
       if (message.includes('error'))
-        this.MessageType = 'danger';
+        this.MessageType = message.messageType;
       else
```
Comment (AbdulSir) [inline frontend/src/views/pages/inventory/Inventory.vue:82]: These if statements do not make sense. Please revise this code.

**Review by AbdulSir — CHANGES_REQUESTED** [review]
All looks good to me. I have tested this PR and works as expected. However, there's a comment about unnecessary use of if statements

File: frontend/src/modules/vehicles/VehicleLocation.vue [hunk 2]
```diff
@@ -4,14 +4,106 @@
       <CCardBody>
         <h4>Location</h4>
         <hr />
+        <GmapMap
+          :center="center"
+          :zoom="11"
+          style="height: 400px"
+        >
+          <GmapInfoWindow :options="infoOptions" :position="infoWindowPos" :opened="infoWinOpen" @closeclick="infoWinOpen=false">
+            <CLink :href="infoLink" target="_blank">{{infoContent}}</CLink>
```
Comment (TZnoj) [inline frontend/src/modules/vehicles/VehicleLocation.vue:13]: In line 13 in VehicleLocation, there is a security hotspot as we are vulnerable to XSS attacks due to bypassing the auto-escaping Vue features by not having the href inside a child node.

File: frontend/src/modules/vehicles/VehicleComments.vue [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+          </CCol>
+        </CCollapse>
+        <hr />
+        <CCol v-if="comments.length > 0">
+          <div class="pt-2" v-for="(comment, index) in comments" :key="comment._id">
+            <CRow class="ml-0 mr-0 d-flex justify-content-between">
+              <h6>{{ comment.author.first_name }}&nbsp;{{ comment.author.last_name }}</h6>
+              <p><small>{{ comment.timestamp }}</small></p>
+            </CRow>
+            <div v-html="comment.comment">
```
Comment (TZnoj) [inline frontend/src/modules/vehicles/VehicleComments.vue:31]: In line 31 in VehicleComments, there is a security hotspot as we are vulnerable to XSS attacks due to bypassing the auto-escaping Vue features by not having the v-html inside a child node.

File: frontend/src/views/pages/inventory/Inventory.vue (hunk 1 above)
Comment (purplesmurf1998) [reply frontend/src/views/pages/inventory/Inventory.vue]: Addressed in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/57/commits/0af7803e90afb19f20de7e446464ad1fcf5405f1)

**Review by AbdulSir — APPROVED** [review]
This looks better now. You may merge. We need to test develop branch after we merge this one.

File: frontend/src/modules/vehicles/VehicleLocation.vue (hunk 2 above)
Comment (purplesmurf1998) [reply frontend/src/modules/vehicles/VehicleLocation.vue]: Line 12 - 14 have been removed since they don't add anything to the placeholder code for now.

**Review by TZnoj — APPROVED** [review]
The SonarCloud issues have been resolved. Approved for merging

**Pull request merged**